### PR TITLE
Social login V2

### PR DESCRIPTION
**Pull request by cilim**
Resolves https://github.com/stormpath/stormpath-rails/issues/81

### CODE REVIEW

**Comment by cilim** [comment]
This feature depends on https://github.com/stormpath/stormpath-sdk-ruby/pull/199

When social login v2 in the SDK is merged to master and released, update the dependency and then merge this into master

File: app/services/stormpath/rails/social_authorization.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+      def generate_url(directory)
+        URI.unescape(URI::HTTPS.build(
+          host: Stormpath::Rails.config.web.client_api.domain_name,
+          path: '/authorize',
+          query: {
+            response_type: 'stormpath_token',
+            account_store_href: directory.href,
+            scope: directory.provider.scope.join('+'),
+            redirect_uri: "#{base_url}#{Stormpath::Rails.config.web.callback.uri}"
```
Comment (cilim) [inline app/services/stormpath/rails/social_authorization.rb]: Add organization_name_key if multitenancy enabled and current_organization is resolved.

File: app/helpers/social_helper.rb [hunk 2]
```diff
@@ -11,6 +11,44 @@ def input_class
     social_providers_present? ? 'col-sm-12' : 'col-sm-8'
   end
 
+  def social_login_v2
```
Comment (DamirSvrtan) [inline app/helpers/social_helper.rb]: `social_login_v2_enabled?`

File: app/services/stormpath/rails/social_authorization.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
+module Stormpath
+  module Rails
+    class SocialAuthorization
```
Comment (DamirSvrtan) [inline app/services/stormpath/rails/social_authorization.rb]: `SocialLoginUrlBuilder.call`

File: app/services/stormpath/rails/social_authorization.rb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            scope: directory.provider.scope.join('+'),
+            redirect_uri: "#{base_url}#{Stormpath::Rails.config.web.callback.uri}"
+          }.to_query
+        ).to_s)
+      end
+
+      def social_directory?(account_store)
+        account_store.class == Stormpath::Resource::Directory &&
+          account_store.provider.respond_to?(:client_id)
+      end
```
Comment (DamirSvrtan) [inline app/services/stormpath/rails/social_authorization.rb]: `SocialDirectories.for(application)`

```ruby
class SocialDirectories
   def self.for(application)
       new(application).call
   end

   def initialize(application)
      @application = application
    end

    def call
        # fetch all social directories from 
     end

     private

    attr_reader :application
end
```

**Pull request merged**